### PR TITLE
(maint) This commit tidy's up the examples and markdown in the README

### PR DESCRIPTION
**Pull request by ajroetker**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,27 +140,37 @@ Upgrading
...
 For example if your config looked like this before:
-~~~ruby
-class {'puppetdb':
-  puppetdb_version => '2.3.5-1.el7',
-}
-class { 'puppetdb::master::config': }
-~~~
+
+    class {'puppetdb':
+    puppetdb_version => '2.3.5-1.el7',
```
Comment (wkalt) [inline README.md]: lost an indent on this one and the next

File: README.md [hunk 2]
```diff
@@ -177,39 +183,66 @@ For this release, all dependency versions have been bumped to their latest. Sign
...
+  it directly.
+* The class `puppetdb::server::jetty_ini` and `puppetdb::server::database_ini`
+  have been marked as private, do not use it directly.
 
 ###Upgrading from 1.x to version 2.x
 
-A major dependency has been changed, so now when you upgrade to 2.0 the dependency `cprice404/inifile` has been replaced with `puppetlabs/inifile`. This may interfer with other modules as they may depend on the old `cprice404/inifile` instead, so upgrading should be done with caution. Check that your other modules use the newer `puppetlabs/inifile` module as interoperation with the old `cprice404/inifile` module will no longer be supported by this module.
+A major dependency has been changed, so now when you upgrade to 2.0 the
+dependency `cprice404/inifile` has been replaced with `puppetlabs/inifile`. This
+may interfer with other modules as they may depend on the old
```
Comment (wkalt) [inline README.md]: interfere

File: README.md [hunk 3]
```diff
@@ -177,39 +183,66 @@ For this release, all dependency versions have been bumped to their latest. Sign
...
+`cprice404/inifile` instead, so upgrading should be done with caution. Check
+that your other modules use the newer `puppetlabs/inifile` module as
+interoperation with the old `cprice404/inifile` module will no longer be
+supported by this module.
 
-Depending on how you install your modules, changing the dependency may require manual intervention. Double check your modules contains the newer `puppetlabs/inifile` after installing this latest module.
+Depending on how you install your modules, changing the dependency may require
+manual intervention. Double check your modules contains the newer
```
Comment (wkalt) [inline README.md]: modules contain

File: README.md [hunk 4]
```diff
@@ -386,70 +473,90 @@ init config file will reflect only what is passed via the `java_args` param.
 
 ####`max_threads`
 
-Jetty option to explicitly set max-thread. The default is undef, so the PuppetDB-jetty default is used.
+Jetty option to explicitly set max-thread. The default is `undef`, so the
```
Comment (wkalt) [inline README.md]: the jetty setting is maxThreads; probably want this plural

File: README.md [hunk 5]
```diff
@@ -386,70 +473,90 @@ init config file will reflect only what is passed via the `java_args` param.
...
+PuppetDB-jetty default is used.
 
 ####`read_database`
 
-Which database backend to use for the read database; Currently only supports `postgres` (default). This option is supported in PuppetDB >= 1.6.
+Which database backend to use for the read database; Currently only supports
```
Comment (wkalt) [inline README.md]: semicolon should be a period, and "Currently" is not needed.

File: README.md [hunk 6]
```diff
@@ -386,70 +473,90 @@ init config file will reflect only what is passed via the `java_args` param.
...
 
-This sets the number of seconds before an SQL query to the read database is considered "slow." Slow SQL queries are logged as warnings, to assist in debugging and tuning. Note PuppetDB does not interrupt slow queries; it simply reports them after they complete.
+This sets the number of seconds before an SQL query to the read database is
+considered "slow." Slow SQL queries are logged as warnings, to assist in
+debugging and tuning. Note PuppetDB does not interrupt slow queries; it simply
+reports them after they complete.
 
-The default value is 10 seconds. A value of 0 will disable logging of slow queries. This option is supported in PuppetDB >= 1.6.
+The default value is 10 seconds. A value of 0 will disable logging of slow
+queries. This option is supported in PuppetDB >= 1.6.
```
Comment (wkalt) [inline README.md]: I don't believe that this works -- I've never seen PDB log a slow statement.

File: README.md [hunk 7]
```diff
@@ -386,70 +473,90 @@ init config file will reflect only what is passed via the `java_args` param.
...
 
 ####`read_conn_max_age`
 
-The maximum time (in minutes), for a pooled read database connection to remain unused before it is closed off.
+The maximum time (in minutes), for a pooled read database connection to remain
```
Comment (wkalt) [inline README.md]: comma not needed

File: README.md [hunk 8]
```diff
@@ -386,70 +473,90 @@ init config file will reflect only what is passed via the `java_args` param.
...
+unused before it is closed off.
 
 If not supplied, we default to 60 minutes. This option is supported in PuppetDB >= 1.6.
 
 ####`read_conn_keep_alive`
 
-This sets the time (in minutes), for a read database connection to remain idle before sending a test query to the DB. This is useful to prevent a DB from timing out connections on its end.
+This sets the time (in minutes), for a read database connection to remain idle
```
Comment (wkalt) [inline README.md]: same

File: README.md [hunk 9]
```diff
@@ -506,9 +621,18 @@ The `puppetdb::server` class manages the puppetdb server independently of the un
...
+The `puppetdb::master::config` class directs your puppet master to use PuppetDB,
+which means that this class should be used on your puppet master node. It’ll
+verify that it can successfully communicate with your puppetdb server, and then
+configure your master to use PuppetDB.
 
-Using this class involves allowing the module to manipulate your puppet configuration files; in particular: puppet.conf and routes.yaml. The puppet.conf changes are supplemental and should not affect any of your existing settings, but the routes.yaml file will be overwritten entirely. If you have an existing routes.yaml file, you will want to take care to use the manage_routes parameter of this class to prevent the module from managing that file, and you’ll need to manage it yourself.
+Using this class involves allowing the module to manipulate your puppet
+configuration files; in particular: puppet.conf and routes.yaml. The puppet.conf
```
Comment (wkalt) [inline README.md]: "Using this class allows the module to manipulate the puppet configuration files puppet.conf and routes.yaml."

File: README.md [hunk 10]
```diff
@@ -519,43 +643,54 @@ Using this class involves allowing the module to manipulate your puppet configur
...
-If you set this, you probably need to set puppetdb_port to match the HTTP port of the PuppetDB.
+If true, use plain HTTP to talk to PuppetDB. Defaults to the value of
+`disable_ssl` if PuppetDB is on the same server as the Puppet Master, or else
+false. If you set this, you probably need to set puppetdb_port to match the HTTP
+port of the PuppetDB.
 
 ####`puppetdb_soft_write_failure`
 
-Boolean to fail in a soft-manner if PuppetDB is not accessable for command submission (defaults to false).
+Boolean to fail in a soft-manner if PuppetDB is not accessable for command
```
Comment (wkalt) [inline README.md]: soft manner, accessible

File: README.md [hunk 11]
```diff
@@ -594,7 +739,10 @@ The `puppetdb::database::postgresql` class manages a postgresql server for use b
 
 ####`listen_addresses`
 
-The `listen_address` is a comma-separated list of hostnames or IP addresses on which the postgres server should listen for incoming connections. This defaults to `localhost`. This parameter maps directly to postgresql's `listen_addresses` config option; use a `*` to allow connections on any accessible address.
+The `listen_address` is a comma-separated list of hostnames or IP addresses on
+which the postgres server should listen for incoming connections. This defaults
+to `localhost`. This parameter maps directly to postgresql's `listen_addresses`
+config option; use a `*` to allow connections on any accessible address.
```
Comment (wkalt) [inline README.md]: semicolon should be a period.

File: README.md [hunk 12]
```diff
@@ -610,19 +758,24 @@ Sets the password for the database user above. Defaults to `puppetdb`.
 
 ####`manage_server`
 
-Conditionally manages the PostgresQL server via `postgresql::server`. Defaults to `true`. If set to false, this class will create the database and user via `postgresql::server::db` but not attempt to install or manage the server itself.
+Conditionally manages the PostgresQL server via `postgresql::server`. Defaults
```
Comment (wkalt) [inline README.md]: PostgreSQL

File: README.md [hunk 13]
```diff
@@ -610,19 +758,24 @@ Sets the password for the database user above. Defaults to `puppetdb`.
...
 ####`test_url`
 
-The URL to use for testing if the PuppetDB instance is running. Defaults to `/pdb/meta/v1/version`.
+The URL to use for testing if the PuppetDB instance is running. Defaults to
+`/pdb/meta/v1/version`.
 
 ####`manage_package_repo`
 
-If this is true, the official postgres.org repo will be added and postgres won't be installed from the regular repository. This setting defaults to `true`.
+If `true`, the official postgres.org repo will be added and postgres won't
```
Comment (wkalt) [inline README.md]: they have a redirect, but the site is postgresql.org

File: README.md [hunk 14]
```diff
@@ -610,19 +758,24 @@ Sets the password for the database user above. Defaults to `puppetdb`.
...
+be installed from the regular repository. This setting defaults to `true`.
 
 ####`postgres_version`
 
-If the postgres.org repo is installed, you can install several versions of postgres. This currently defaults to 9.4 which is the latest stable version.
+If the postgres.org repo is installed, you can install several versions of
```
Comment (wkalt) [inline README.md]: ditto

**Comment by wkalt** [comment]
left some minor notes. this generally looks good to me. I'd also change occurrences of "puppetdb" to "PuppetDB" when we're not in a code context, "puppet master" and "Puppet Master" should be "Puppet master", and "Currently" is almost never necessary in here IMHO.

**Comment by ajroetker** [comment]
@wkalt cleaned up for your comments!

**Comment by kbarber** [comment]
@wkalt @ajroetker are we good for merge?

**Pull request merged**